File: week6/pset6/readability/readability.py
from cs50 import get_string


def main():
    # define variables
    letters = 0
    words = 1  # number of words = number of spaces +1
    sentences = 0
    
    # get user input
    text = get_string("Text: ")
    # number of chars (inclusive of words, whitespaces and symbols)
    char = len(text)
    
    for i in range(char):
        if text[i].isalpha():
            letters += 1
        
        if text[i].isspace():
            words += 1
        
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1
        
    # Coleman-Liau index
    index = round(0.0588 * ((letters / words) * 100) - 0.296 * ((sentences / words) * 100) - 15.8)
    
    # print grade level
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else: 
        print(f"Grade {index}")


main()

File: week2/pset2/readability/readability.c
#include <ctype.h>
#include <math.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // get input from user
    string text = get_string("Text: ");
    
    // Print number of letters
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    
    // Print number of words ( only those separated by a space to be counted)
    int words = 1; // no. of words = no. of spaces + 1
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words ++;
        }
    }
    
    // Print out number of sentences (./?/! means one sentence)
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    
    // Grade the text ( index = 0.0588 * L - 0.296 * S - 15.8) & round
    double L = ((float)letters / (float)words) * 100;
    double S = ((float)sentences / (float)words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    // Print the grade out
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}